internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -274,56 +285,82 @@ func convertVolumes(input map[string]manifest.Volume) ([]*template.Volume, error
 	}
 	output := []*template.Volume{}
 	for name, volume := range input {
-		// Set default values correctly.
-		fsID := volume.EFS.FileSystemID
-		if aws.StringValue(fsID) == "" {
-			return nil, errNoFSID
-		}
-
-		rootDir := volume.EFS.RootDirectory
-
-		// Validate that root directory path doesn't contain spaces or shell injection.
-		if err := validateRootDirPath(aws.StringValue(rootDir)); err != nil {
-			return nil, fmt.Errorf("validate root directory path %s: %w", aws.StringValue(rootDir), err)
-		}
+		// Volumes can contain either:
+		//   a) an EFS configuration, which must be valid
+		//   b) no EFS configuration, in which case the volume is created using task scratch storage in order to share
+		//      data between containers.
 
-		if aws.StringValue(rootDir) == "" {
-			rootDir = aws.String(defaultRootDirectory)
+		// If there is no EFS information, just add the Name to the volume.
+		if volume.EFS == nil {
+			v := template.Volume{
+				Name: aws.String(name),
+			}
+			output = append(output, &v)
+			continue
 		}
```

Comment:
I feel we should handle this logic in `convertEFSConfiguration` instead: if the input is `nil` then return `nil`